app/models/item.rb

```diff
@@ -25,4 +25,23 @@ def no_orders?
     item_orders.empty?
   end
 
+  def self.top_5
+    top_5_data = ItemOrder.find_by_sql ["SELECT items.name, item_id, SUM(quantity) AS total_purchased FROM item_orders JOIN items ON item_orders.item_id = items.id GROUP BY item_id, items.name ORDER BY total_purchased DESC LIMIT 5"]
```

Comment:
This is pure SQL. I gave up on AR. Figured this out using Postico and thanks to a little helpful hint from Mike. Feeling lukewarm about succeeding with something so ugly.

Reply:
You put in WORK for this, it may not look the prettiest, but it functions, and that rocks my socks.

Reply:
![](https://media2.giphy.com/media/VexReXhmpp3m5QXZCT/giphy.gif?cid=5a38a5a28ab705b021b079b03b9785c3bbac958f4c285f03&rid=giphy.gif)